plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -268,6 +253,38 @@ private boolean validatePage() {
     return true;
   }
 
+  private boolean validateLocation(String location) {
+    if (location.isEmpty()) {
+      setMessage(Messages.getString("PROVIDE_LOCATION"), INFORMATION); //$NON-NLS-1$
+      return false;
+    } else {
+      try {
+        java.nio.file.Path path = Paths.get(location);
+        do {
+          if (Files.exists(path)) {
+            if (!Files.isDirectory(path)) {
+              String message = MessageFormat.format(Messages.getString("FILE_LOCATION"), path.toString()); //$NON-NLS-1$
+              setMessage(message, WARNING);
+              return false;              
+            } else if (!Files.isWritable(path)) {
+              String message = MessageFormat.format(Messages.getString("NONWRITABLE"), location); //$NON-NLS-1$
+              setMessage(message, WARNING);
+              return false;
+            } else { // writable directory
+              return true;
+            }
+          }
+          path = path.getParent();
+        } while (path != null);
```

Comment:
I don't think that can happen because:

1. Paths.get doesn't return null
2. If Paths.get does return null, a NullPointerException is thrown earlier